xdsl/dialects/snitch_stream.py

```diff
@@ -152,25 +152,31 @@ def simplified(self) -> StridePattern:
         """
         if len(self.ub) < 2:
             return self
-        d = -2
+
+        tuples = tuple(
+            (bound.data, stride.data)
+            for bound, stride in zip(self.ub.data, self.strides.data)
+            # Exclude singe iteration bounds
+            if bound.data != 1
+        )
+
+        d = len(tuples) - 2
```

Comment:
Paradoxically, this `d` index and its initialization irks me a lot.
Can you please add a simple comment on what this index is, and why it is decremented by 2?